### DISCUSSION TITLE
SPF check

### DISCUSSION THREAD

**Discussion by PCL-CrisKolkman** [postalserver/postal#1927] [Help with using Postal]
Hello,

Is there any way to make Postal do a (correct) SPF check?
All the DNS records for my domain are correct but when a spammer connects to my Postal server and sends a mail from me@myowndomain.com to me@myowndomain.com, Postal just accepts the message and forwards it to my Exchange server while I'm expecting the message would get blocked at this point.

Am I missing something here?

**Comment by PCL-CrisKolkman** [comment]
Here is an example for such a message.
Both the **from** and **to** are from my own domain.

![image](https://user-images.githubusercontent.com/97805195/160081073-ab3c8c75-3634-433e-be23-57c024e2eca3.png)

When I lookup the IP address being used to send from, it's also blacklisted:

![image](https://user-images.githubusercontent.com/97805195/160081197-0fea5e93-294d-425d-918b-b4eb50ae070c.png)

Why are these images being accepted?

**Comment by willpower232** [comment]
Postal is designed to accept any message passed to it and send it on. The only SPF check it cares about is trying to make sure the domain it uses trusts Postal to help ensure the email will be delivered successfully.

If you have a public form which allows any email to be sent, you need to restrict that form so it only sends to a specific address and you could also use a captcha to filter out bots.

You can enable spamassassin or rspamd within Postal to try and prevent these messages being sent but you'll need to install spamassassin or rspamd yourself.

**Reply by PCL-CrisKolkman** [reply]
> Postal is designed to accept any message passed to it and send it on. The only SPF check it cares about is trying to make sure the domain it uses trusts Postal to help ensure the email will be delivered successfully.
> 
> If you have a public form which allows any email to be sent, you need to restrict that form so it only sends to a specific address and you could also use a captcha to filter out bots.
> 
> You can enable spamassassin or rspamd within Postal to try and prevent these messages being sent but you'll need to install spamassassin or rspamd yourself.

Okay thank you, I won't use Postal for incoming messages anymore then because spamassassin is also not preventing that those messages are coming in :)
There are no forms being used, these are direct connections to the Postal server from spammers and Postal should do an SPF/blacklist check but it doesn't.
